tests/test_core.py

```diff
@@ -682,6 +692,36 @@ def test_context_finalizer(tCWFstatMapVersion, transientWindowType, cleanup):
     assert detstat > 0
 
 
+def test_atoms_io(tmp_path, CFS_default_params, lambda_params):
+    search = pyfstat.ComputeFstat(
+        **CFS_default_params,
+        computeAtoms=True,
+    )
+    search.get_fullycoherent_detstat(**lambda_params)
+    atoms_orig = pyfstat.tcw_fstat_map_funcs.reshape_FstatAtomsVector(
+        search.FstatResults.multiFatoms[0].data[0]
+    )
+    atomsdir = tmp_path
+    search.write_atoms_to_file(os.path.join(atomsdir, "CFS"))
+    atomsfiles = [
+        os.path.join(atomsdir, f) for f in os.listdir(atomsdir) if "Fstatatoms" in f
+    ]
```

Comment:
```suggestion
    atomsdir = str(tmp_path)
    search.write_atoms_to_file(os.path.join(atomsdir, "CFS"))
    atomsfiles = [
        os.path.join(atomsdir, f) for f in os.listdir(atomsdir) if "Fstatatoms" in f
    ]
```
`tmp_path` is a `Path` object, so you may want to cast it into string to avoid a headache.

Reply:
Otherwise, fully embrace `Path` and see if it works:
```suggestion
    atomsdir = tmp_path
    search.write_atoms_to_file(atomsdir ( "CFS")
    atomsfiles = [
        (atomsdir / f) for f in os.listdir(atomsdir) if "Fstatatoms" in f
    ]
```

Reply:
It still confuses me that they made this non-portable-looking syntax the portable default, but sure, let's go for it.

Fixed one `(` to `/`.

Reply:
Careful: I'm not saying this will work, I'm saying this *may* work if our code doesn't require too much "stringness" from `Path` :wink: